javatests/arcs/core/entity/VariableEntityBaseTest.kt

```diff
@@ -0,0 +1,95 @@
+package arcs.core.entity
+
+import arcs.core.crdt.VersionMap
+import arcs.core.storage.testutil.DummyStorageKey
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.test.assertFailsWith
+
+
+@RunWith(JUnit4::class)
+@Suppress("UNCHECKED_CAST")
+class VariableEntityBaseTest {
+
+    private lateinit var entity: DummyVariableEntity
+
+    @Before
+    fun setUp() {
+        SchemaRegistry.register(DummyEntity.SCHEMA)
+        SchemaRegistry.register(DummyVariableEntity.SCHEMA)
+        entity = DummyVariableEntity()
+    }
+
+    @After
+    fun tearDown() {
+        SchemaRegistry.clearForTest()
+    }
+
+    @Test
+    fun serializationRoundTrip() {
+        val biggerEntity = DummyEntity()
+            .apply {
+                text = "abc"
+                num = 12.0
+                bool = true
+                ref = createDummyReference("foo")
+                texts = setOf("aa", "bb")
+                nums = setOf(1.0, 2.0)
+                bools = setOf(true, false)
+                refs = setOf(createDummyReference("ref1"), createDummyReference("ref2"))
+            }
+
+        val biggerRaw = biggerEntity.serialize()
+
+        val variableEntity = DummyVariableEntity()
+        variableEntity.deserializeForTest(biggerRaw)
+
+        val backToBiggerRaw = variableEntity.serialize()
+        assertThat(backToBiggerRaw).isEqualTo(biggerRaw)
+
+        val backToBigger = DummyEntity()
+        backToBigger.deserializeForTest(backToBiggerRaw)
+        assertThat(backToBigger).isEqualTo(biggerEntity)
+    }
+
+    @Test
+    fun variableSerialization_propertyAccess() {
+        val biggerEntity = DummyEntity()
```

Comment:
Do you want to extract that from 2 tests into a shared method?

Reply:
That sounds good to me. I've moved it to the startUp method.